com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs

```diff
@@ -1251,6 +1251,25 @@ void SetupVignette(Material material)
                 m_Vignette.smoothness.value * 5f
             );
 
+#if ENABLE_VR && ENABLE_XR_MODULE
```

Comment:
Maybe move the XR specific block into a function, easier to read than bunch of defines. 
That function could even be in xrPass. 
Something like
vec4 xrPass.applyXREyeOffset(vec2 center); which packs XR eye data into Vec4. Or xrPass.applyEyeOffset( ref vec2 centerAsLeft, out vec2 right);

But just moving this block to SetupVignetteXR() would be tiny bit better IMO.

Reply:
I think we can extract a good amount of code to the XRPass class, but we still need to do some part of the work in SetupVignette() since when we are rendering in Multi-pass (each eye are rendered completely separately), the center still need to be corrected for each eyes and then placed in v2.xy since AFAIK the shader does not have a variation for "XR enabled, but not single-pass". 

I'll see what I can do to reduce the amount of XR specific code in here.